src/Search.jl

```diff
@@ -0,0 +1,55 @@
+export pkgsearch
+
+# recursive function for applying search criteria
+function keycheck(data::Dict{<:Any,<:Any},str::Array{String,1},mode::Array{Symbol,1})
+    found = false
+    for key in keys(data)
+        if :deps in mode
+            for s in str
+                (key == s) && (found = true)
+            end
+        end
+        if (((:any in mode) && (typeof(data[key]) == String)) ||
+            ((:name in mode) && (key == "name")) ||
+            ((:keywords in mode) && (key == "keywords")) ||
+            ((:desc in mode) && (key == "description")))
+            for s in str
+                contains(data[key],s) && (found = true)
+            end
+        end
+        if typeof(data[key]) <: Dict{<:Any,<:Any}
+            keycheck(data[key],str,mode) && (found = true)
+        end
+    end
+    return found
+end
+
+# directory search for package toml
+function pkgsearch(mode::Array{Symbol,1},str::Array{String,1})
+    m = mode
+    (:any in) m && push!(m,:deps,:name)
+    pkglist = Array{String}(uninitialized,0)
```

Comment:
Any empty vector doesn't need to be uninitialized since there's no elements. You can just do `String[]` here. You also may want to keep track of UUIDs and names since names are no longer unique.